### DISCUSSION TITLE
exclude routes

### DISCUSSION THREAD

**Discussion by tasiotas** [ijkml/nuxt-umami#23] [Q&A] [answered]
Hi,

Is there a way to exclude tracking on specific routes?
For example, do not track users on /dashboard/* or /account*

Thank you

**Comment by ijkml** [comment]
Hi,

To achieve that, I think you will need to disable Umami auto-track (set `autoTrack` to `false` in the module options). Then use `Umami.trackView` and Nuxt's `RouteMiddleware`. This should give you a fine-grained control of routes you want and don't want to track.

See Nuxt route-middleware: https://nuxt.com/docs/api/utils/define-nuxt-route-middleware

I have not used it this way, but it should work. Hopefully 😅. And if you need help implementing this, I'll be super happy to help.

**Reply by tasiotas** [reply]
thank you for the idea.
I will just do whitelisting style implementation, and put `Umami.trackView` in views that I want to track.

**Reply by ijkml** [reply, marked as answer]
Hey @tasiotas 
If you're still interested in that idea, please take a look at https://github.com/ijkml/nuxt-umami-nuxt3/blob/7f27c8b2b11b06a99c8b6b3b6f5f2f2c0c0f7144/middleware/umami.global.ts

**Reply by tasiotas** [reply]
thank you!

**Reply by ijkml** [reply]
Glad I could help.

Make sure to disable `autoTrack` 🚀
